### PR TITLE
Prep for alignment with Flutter analysis options

### PR DESCRIPTION
**Pull request by stuartmorgan**
Renames the old analysis_options.yaml to analysis_options_legacy.yaml,
replacing it with a slightly modified copy of flutter/flutter's analysis
options.

Each plugins has a temporary local analysis_options.yaml that points to
the legacy version. This allows for inceremental conversion on a
per-plugin basis, which should make the problem more tractable.

Since this hasn't yet been enabled for any packages, it's likely that as
it is we'll find a few local modification we need to make to the root
analysis_options (e.g., things that conflict with 'dart format').

Part of https://github.com/flutter/flutter/issues/76229

## Pre-launch Checklist

- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,13 +1,240 @@
-include: package:pedantic/analysis_options.1.8.0.yaml
+# This is a copy (as of March 2021) of flutter/flutter's analysis_options file,
```
Comment (ditman) [inline analysis_options.yaml:1]: Note that if you took this from the root of the flutter repo, it gets "extended" in its subdirectories:

* [flutter/packages](https://github.com/flutter/flutter/blob/master/packages/analysis_options.yaml)
* [flutter/packages/flutter](https://github.com/flutter/flutter/blob/master/packages/flutter/analysis_options.yaml)

(I'm not sure which exact flavor is being used when we get tested, but I think this file at least needs the `no_runtimeType_toString` linter rule enabled.

**Review by ditman — APPROVED** [review]
Let's go! I'd mark the Plus plugins with a comment in their .yaml so we leave those for last.

File: analysis_options.yaml (hunk 1 above)
Comment (stuartmorgan) [reply analysis_options.yaml:1]: Good call, added the two extra rules in a marked section..

**Comment by ditman** [comment]
This change is breaking the rolls into flutter/flutter:

```
Activated flutter_plugin_tools 0.0.45+3.
▌09:47:03▐ ELAPSED TIME: 5.143s for ../../../k/flutter/bin/cache/dart-sdk/bin/pub global activate flutter_plugin_tools in ../../tmp/t/pluginsDYZTNQ
▌09:47:03▐ RUNNING: cd ../../tmp/t/pluginsDYZTNQ; ../../../k/flutter/bin/cache/dart-sdk/bin/pub global run flutter_plugin_tools analyze
Verifying analysis settings...
Found an extra analysis_options.yaml in /b/s/w/ir/tmp/t/pluginsDYZTNQ/packages/android_alarm_manager/analysis_options.yaml.
If this was deliberate, pass the package to the analyze command with the --custom-analysis flag and try again.
```

We should let the analysis command that all our plugins have a custom `analysis_options.yaml` for now?

Investigating.

**Comment by ditman** [comment]
The change needs to happen around here:

https://github.com/flutter/flutter/blob/2e06ed078726e31c5afca36ae9c6736060c7b7e8/dev/bots/test.dart#L851-L860

**Pull request merged**